### PR TITLE
Address issues 561 and 307

### PR DESCRIPTION
**Pull request by dvandyk**

### CODE REVIEW

File: eos/utils/exception.cc [hunk 1]
```diff
@@ -19,17 +19,109 @@
...
+
+    Context::Context(const std::string & entry, const std::source_location location)
+    {
+        impl::context.push_back(std::make_tuple(location, entry));
+    }
+
+    Context::Context(const Context & other) = default;
+
+    constexpr Context &
+    Context::operator= (const Context & other) = default;
```
[the remark this answers is not in the document]
Comment (dvandyk) [reply eos/utils/exception.cc:39]: Would be synthesized anyway. This way i'm being explicit!

File: .github/workflows/ubuntu-build+check+deploy.yaml [hunk 2]
```diff
@@ -18,13 +18,12 @@ jobs:
...
                 cfg:
-                    - { os: focal,    cxx: g++,          cxxflags: '-O2 -g -march=x86-64' }
-                    - { os: focal,    cxx: clang++-12,   cxxflags: '-O2 -g -march=x86-64' }
                     - { os: jammy,    cxx: g++,          cxxflags: '-O2 -g -march=x86-64' }
+                    - { os: jammy,    cxx: clang++-14,   cxxflags: '-O2 -g -march=x86-64' }
         name: Build on ${{ matrix.cfg.os }} using ${{ matrix.cfg.cxx }}
         runs-on: ubuntu-20.04
         container:
-            image: eoshep/ubuntu-${{ matrix.cfg.os }}:latest
+            image: eoshep/ubuntu-${{ matrix.cfg.os }}:ae4ca8649bf120510f83ff13b95ef2c0abc49e70
```
Comment (mreboud) [inline .github/workflows/ubuntu-build+check+deploy.yaml:26]: So here you are fixing a specific version, right? Why?

Comment (dvandyk) [reply .github/workflows/ubuntu-build+check+deploy.yaml:26]: a) Versioning of the image is prudent if you make breaking changes. I should have done that before.
b) In this specific case, I made two successive changes to the docker images, but they would have broken ongoing workflows.

Comment (dvandyk) [reply .github/workflows/ubuntu-build+check+deploy.yaml:26]: It is the full SHA1 id of the commit that produced the docker image over in eos/docker.io.

Comment (mreboud) [reply .github/workflows/ubuntu-build+check+deploy.yaml:26]: Ok ok, should this be documented somewhere? E.g. this should be adapted when pushing a new release, no?

Comment (dvandyk) [reply .github/workflows/ubuntu-build+check+deploy.yaml:26]: I just added it to the wiki: https://github.com/eos/eos/wiki/How-to-bump-the-docker-images

**Review by mreboud — APPROVED** [review]
Cool, thanks @dvandyk!

**Pull request merged**